internal/cache/util/util.go

```diff
@@ -99,3 +101,25 @@ func IsCacheHandleInvalid(readErr error) bool {
 		strings.Contains(readErr.Error(), ErrInSeekingFileHandleMsg) ||
 		strings.Contains(readErr.Error(), ErrInReadingFileHandleMsg)
 }
+
+func CheckCacheDirectoryPermissions(dirPath string) (bool, error) {
+	_, statErr := os.Stat(dirPath)
+
+	if statErr != nil {
+		err := os.MkdirAll(dirPath, FileDirPerm)
+		if err != nil {
+			return false, fmt.Errorf("error in creating directory structure %s: %v", dirPath, err)
+		}
+		return true, nil
+	}
+
+	f, err := fsutil.AnonymousFile(dirPath)
+	f.Close()
```

Comment:
should we check the error returned from f.Close() as well ?

Reply:
my understanding was even if close returns an error, we would not want to fail mounting hence ignored it, let me know what should be our action for this error

Reply:
i think we should return error because it signals that something wrong happened while checking if directory is writable or not.

Reply:
Can you please validate my understanding...If any error occurs while creating annonymous we are already returning error and code panics, in addition to that now we also return error and panic if any error is returned from f.close() i.e. closing the annonymous file handle ? Also verified in temp dir we are ignoring this error
https://github.com/GoogleCloudPlatform/gcsfuse/blob/92dd6ae9d6feddc95397764fd678e5703de8bff3/mount.go#L50C3-L50C12

Reply:
So in general, we shouldn't ignore any error  because that can potentially mask bugs/problems (e.g. https://stackoverflow.com/questions/62594017/error-handing-in-go-when-to-ignore-errors)
So in the above case, either we should change f.close() method itself to not throw error at all or handle the error on our side.

Reply:
got it... have added changes to handle error in latest commit... ready for final review !